react/lib/components/Widget/Widget.tsx

```diff
@@ -526,31 +504,29 @@ export const Widget: React.FunctionComponent<WidgetProps> = props => {
 
   const handleQrCodeClick = (): void => {
     if (disabled || to === undefined) return;
-    const address = to;
-    let thisAmount: number | undefined;
+    if (!url || !copy(url)) return;
+    setCopied(true);
+    setRecentlyCopied(true);
+  };
 
-    if (convertedCurrencyObj) {
-      thisAmount = convertedCurrencyObj.float;
-    } else {
-      thisAmount = thisCurrencyObject ? thisCurrencyObject.float : undefined;
-    }
-    if (isValidCashAddress(address)) {
-      prefixedAddress = `bitcoincash:${address.replace(/^.*:/, '')}`;
-    } else {
-      prefixedAddress = `ecash:${address.replace(/^.*:/, '')}${
-        thisAmount ? `?amount=${thisAmount}` : ''
-      }`;
+  const resolveUrl = (currency: string, amount?: number): string | undefined => {
```

Comment:
You should use [useCallback](https://react.dev/reference/react/useCallback#usecallback) here: notice how this function use reactive values like `disabled`, `to`, and `opReturn`.

I actually didn't find issues when I tested, but I think they could appear if we don't address that